### PR TITLE
Move sequence and merge-tree to new applyStashedOps model, and add fuzz

### PR DESCRIPTION
**Pull request by anthony-murphy**
## Background
This change is related to https://github.com/microsoft/FluidFramework/pull/19518. That change introduced a new and simpler mechanism for applying stash ops. Specifically, it allows dds to reuse their existing flows for creating local changes, rather than needing custom flows to apply the change and get the local op metadata. This reduces complexity, bundle size, and bug potential by having fewer code paths to do the same thing; apply changes locally.

## Overview
This change moves merge-tree, sequence, and interval collections to the new applyStashedOps  model. 
Beyond that i've done some clean up in the interval collection code to clean up a few things which made making the change easier:

- removed redundant constants for reservedIntervalIdKey
- collapsed messageHandlers to the single existing messageHandler 
- improved type handling with the use of type guard.
- deleted a bunch of applyStashedOps  code that is no longer necessary

[AB#7145](https://dev.azure.com/fluidframework/internal/_workitems/edit/7145)

### CODE REVIEW

File: packages/dds/sequence/src/defaultMap.ts [hunk 1]
```diff
@@ -117,7 +94,37 @@ export class DefaultMap<T> {
 	/**
 	 * Mapping of op types to message handlers.
 	 */
-	private readonly messageHandlers: ReadonlyMap<string, IMapMessageHandler> = new Map();
+	private readonly messageHandler = {
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/defaultMap.ts:97]: this is just copy pasted from below, and removed from the dictionary, as it always had only one entry, this one

File: packages/dds/sequence/src/defaultMap.ts [hunk 2]
```diff
@@ -27,39 +27,16 @@ import {
...
+	reservedIntervalIdKey,
+} from "./intervalCollection";
 
 function isMapOperation(op: unknown): op is IMapOperation {
 	return typeof op === "object" && op !== null && "type" in op && op.type === "act";
 }
-/**
- * Defines the means to process and submit a given op on a map.
- */
-interface IMapMessageHandler {
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/defaultMap.ts:37]: we don't need this abstraction, as there is only one

File: packages/dds/sequence/src/defaultMap.ts [hunk 3]
```diff
@@ -384,55 +418,6 @@ export class DefaultMap<T> {
 		return new ValueTypeLocalValue(localValue, this.type);
 	}
 
-	/**
-	 * Get the message handlers for the map.
-	 * @returns A map of string op names to IMapMessageHandlers for those ops
-	 */
-	private getMessageHandlers() {
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/defaultMap.ts:391]: moved above

File: packages/dds/sequence/src/defaultMap.ts [hunk 4]
```diff
@@ -327,14 +357,18 @@ export class DefaultMap<T> {
...
+		op: unknown,
 		local: boolean,
-		message: ISequencedDocumentMessage | undefined,
+		message: ISequencedDocumentMessage,
 		localOpMetadata: unknown,
 	): boolean {
-		const handler = this.messageHandlers.get(op.type);
-		if (handler !== undefined) {
-			handler.process(op, local, message, localOpMetadata as IMapMessageLocalMetadata);
+		if (isMapOperation(op)) {
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/defaultMap.ts:365]: no more dictionary. just check if its a map op via existing type guard, and if so, call the handler. eventually we can compress the abstraction here even further, but don't want this pr to introduce too much chrun

File: packages/dds/sequence/src/intervalCollection.ts [hunk 5]
```diff
@@ -115,7 +113,7 @@ export enum Side {
 	After = 1,
 }
 
-const reservedIntervalIdKey = "intervalId";
+export const reservedIntervalIdKey = "intervalId";
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/intervalCollection.ts:116]: this was redefined all over the place, no we just have one copy

File: packages/dds/sequence/src/intervalCollection.ts [hunk 6]
```diff
@@ -198,15 +196,18 @@ export function endpointPosAndSide(
 	};
 }
 
-function toSequencePlace(pos: number | "start" | "end", side: Side): SequencePlace {
-	return typeof pos === "number" ? { pos, side } : pos;
+export function toSequencePlace(
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/intervalCollection.ts:199]: these are reused for apply stashed op, so need to be exported, and support undefined sides

File: packages/dds/sequence/src/sequence.ts [hunk 7]
```diff
@@ -299,7 +299,7 @@ export abstract class SharedSegmentSequence<T extends ISegment>
 
 		this.client.prependListener("delta", (opArgs, deltaArgs) => {
 			const event = new SequenceDeltaEvent(opArgs, deltaArgs, this.client);
-			if (opArgs.stashed !== true && event.isLocal) {
+			if (event.isLocal) {
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/sequence.ts:302]: this is where sequence submits its ops based on events, now we submit all local ops.

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**